### PR TITLE
Computing access and resources

### PR DESCRIPTION
**Pull request by fredericpoitevin**
A first draft to document getting computer accounts and accessing the data. Looking for feedback.

### CODE REVIEW

File: docs/before/accounts.md [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+In order to get access to the data and to submit SLURM jobs, reach an interactive node:
+```
+ssh psana
+```
+Once on the interactive node, execute the following command (or place it in your `~/.bashrc`) in order to set the necessary environment variables and environments that will allow you to find the data and process it:
+``` 
+source /sdf/group/lcls/ds/ana/sw/conda1/manage/bin/psconda.sh
+```
+!!! note "FFB at S3DF"
+    In S3DF, both FFB and Offline storage are accessible. By default, the command above points to the offline storage. In order to access the FFB storage, you can override the environment variable that points to the data:
```
Comment (apeck12) [inline docs/before/accounts.md:78]: Perhaps worth mentioning on what timescale data are expected to migrate from the FFB to offline storage after a beamtime ends?

File: docs/before/accounts.md [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+!!! note "Enabling access to the S3DF"
+    In order to use S3DF, it is necessary to enable your S3DF account using your SLAC UNIX account. Detailed information [in the S3DF documentation here][7].
+
+!!! danger "Access to NERSC"
+    At the moment, users need a NERSC account to run LCLS workflows there. See [this link][9] for some information on how to create an account at NERSC. Work is in progress to remove this unncessary constraint.
+
+# Step 2 - Access and process data
+Now that you can access the computer systems at SLAC, it is time to get yourself oriented. 
+It is useful for that to understand how the data flows from the beamline to storage, *a.k.a* to understand the LCLS Data Systems, built with 3 successive levels:
```
Comment (apeck12) [inline docs/before/accounts.md:22]: Maybe instead reword it to: "For that, it is useful to understand..."

Comment (apeck12) [inline docs/before/accounts.md:22]: Maybe instead: "For that, it is useful to understand..."

File: docs/before/accounts.md [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+- **2 &#x3e;  FFB - Fast Feedback**:<br> 
+provides diagnostics and results within <1min. *(All)*
+- **3 &#x3e;  Offline Analysis**:<br> 
+for intensive compute during and after the beamtime. *(All)*
+
+*TODO: figure to describe the network and data infrastructure.*
+
+!!! danger "FFB migration to S3DF"
+    S3DF is a new capability at SLAC and is currently still a work in progress. 
+    The FFB nodes will soon be a part of it - when that happens, it will be more seamless to switch from online analysis to offline analysis at the end of the beamtime.
```
Comment (apeck12) [inline docs/before/accounts.md:35]: and between shifts as well? I also wonder if it would be useful at some point on this page to note that each experiment is assigned a particular queue on the FFB during the experiment, while anaq should be used between shifts and after?

**Review by apeck12 — APPROVED** [review]
Very cool -- this will be such a useful resource. Thanks, Fred!

File: docs/before/accounts.md [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+    - **Account** - [Create a UNIX account][6]
+    - **Computing** - [Access and Resources][1]
+    - **Data Analysis** - for [LCLS-I][2] and for [LCLS-II][3]
+    - **FFB docs** - [PCDS docs][8]
+    - **S3DF docs** - ["Running at S3DF" for LCLS][4] and [official S3DF documentation][5].
+
+
+# Step 1 - Get a computer account
+
+To access SLAC computing resources, you need a valid SLAC UNIX account. A detailed description of the steps required to create one can be found [here][6].
```
Comment (gadorlhiac) [inline docs/before/accounts.md:12]: Definitely agree we should link to the confluence but maybe it's worth adding a quick synopsis, e.g. something like:
1. Take CS101 (link)
2. Understand and accept the terms of service (link)
3. Apply for an account through the user portal (link)

File: docs/before/accounts.md [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+
+!!! note "Enabling access to the S3DF"
```
Comment (gadorlhiac) [inline docs/before/accounts.md:14]: We should probably add this to the glossary. If I don't know what the S3DF is this might be confusing to me. Why should I care about it?

We can also link to the glossary

File: docs/before/accounts.md (hunk 2 above)
Comment (gadorlhiac) [reply docs/before/accounts.md:22]: +1

File: docs/before/accounts.md [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+
+*TODO: figure to describe the network and data infrastructure.*
+
+!!! danger "FFB migration to S3DF"
+    S3DF is a new capability at SLAC and is currently still a work in progress. 
+    The FFB nodes will soon be a part of it - when that happens, it will be more seamless to switch from online analysis to offline analysis at the end of the beamtime. 
+    But for now (March 2023), the approach described in the first section below is still valid for past experiments.
+    In time, experiments will be analyzed on S3DF both during and after beamtimes (see following section).
+
+### &#x2192; on the FFB - Fast Feedback System {: #ffb}
```
Comment (gadorlhiac) [inline docs/before/accounts.md:39]: Maybe it would be useful to have an overview of available nodes before diving into the specifics of the FFB?

We don't have to explain everything SSH by any means but describing which nodes are entry points and which can be reached from there? I know this will change in the future, but I think your notes/danger signs are good at highlighting that this is in a state of flux .

General idea: a three step guide:
1. Login node
2. SSH to interactive node
3. Source necessary vars

Then a table :
Login nodes: pslogin...
Interactive nodes: psana, psffb...

File: docs/before/accounts.md [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+ssh psffb
+```
+The experiments and their data can be found at the following path:
+```
+/cds/data/drpsrcf/<instrument>/<experiment>
+```
+!!! danger "Data import/export"
+    Note that the FFB nodes are not exposed to external networks. If you need to transfer data, you can go through `psexport.slac.stanford.edu`.
+
+*TODO: a few sentences on batch, interactive and jupyter.*
```
Comment (gadorlhiac) [inline docs/before/accounts.md:56]: Is the intention to incldue sourcing psconda in this section as well? If not I think having that like you do below would be very useful

File: docs/before/accounts.md [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+    In S3DF, both FFB and Offline storage are accessible. By default, the command above points to the offline storage. In order to access the FFB storage, you can override the environment variable that points to the data:
+    ```
+    export SIT_PSDM_DATA=/sdf/data/lcls/drpsrcf/ffb
+    ```
+The experiments and their data can be found at the following path:
+```
+${SIT_PSDM_DATA}/<instrument>/<experiment>
+``` 
+
+*TODO: a few sentences on batch, interactive and jupyter.*
```
Comment (gadorlhiac) [inline docs/before/accounts.md:87]: Reading it now, I think that it will probably be worth having a separate page on SLURM jobs/batch system?

Maybe not, but I feel it may be easier to digest:
- This page = getting an account, understanding nodes, accessing data
- batch page = submitting intensive jobs, basic commands (squeue, scancel etc)

**Review by gadorlhiac — COMMENTED** [review]
Looking good so far, thanks!

I added some comments with a couple of ideas to consider

File: docs/before/accounts.md (hunk 2 above)
Comment (fredericpoitevin) [reply docs/before/accounts.md:22]: Done

File: docs/before/accounts.md (hunk 4 above)
Comment (fredericpoitevin) [reply docs/before/accounts.md:12]: Done

File: docs/before/accounts.md (hunk 5 above)
Comment (fredericpoitevin) [reply docs/before/accounts.md:14]: Done

File: docs/before/accounts.md [hunk 9]
```diff
@@ -0,0 +1,101 @@
...
+
+
+# Step 1 - Get a computer account
+
+To access SLAC computing resources, you need a valid SLAC UNIX account. A detailed description of the steps required to create one can be found [here][6].
+
+!!! note "Enabling access to the S3DF"
+    In order to use S3DF, it is necessary to enable your S3DF account using your SLAC UNIX account. Detailed information [in the S3DF documentation here][7].
+
+!!! danger "Access to NERSC"
```
[the remark this answers is not in the document]
Comment (fredericpoitevin) [reply docs/before/accounts.md:17]: I am removing mention of NERSC for now since it's a edge case that we can always document when the situation arises.

File: docs/before/accounts.md (hunk 8 above)
Comment (fredericpoitevin) [reply docs/before/accounts.md:87]: Good idea.

File: docs/before/accounts.md (hunk 1 above)
Comment (fredericpoitevin) [reply docs/before/accounts.md:78]: Done

File: docs/before/accounts.md (hunk 3 above)
Comment (fredericpoitevin) [reply docs/before/accounts.md:35]: Good idea. Let's follow Gabriel's suggestion and have another page about batch and interactive computing with SLURM and Jupyter.

File: docs/before/accounts.md (hunk 7 above)
Comment (fredericpoitevin) [reply docs/before/accounts.md:56]: Done

File: docs/before/accounts.md (hunk 8 above)
Comment (fredericpoitevin) [reply docs/before/accounts.md:87]: Created Issue #15

File: docs/before/accounts.md [hunk 10]
```diff
@@ -0,0 +1,125 @@
...
+
+
+!!! danger "migration to S3DF"
+    S3DF is a new capability at SLAC and is currently still a work in progress. 
+    New experiments will be stored and processed on S3DF while previous experiments can still be access on the PCDS clusters.
+    On S3DF, it will be more seamless to switch from online analysis to offline analysis between shifts and at the end of the beamtime.
+
+# Step 3 - Access your experiment data
+
+Your experiment data can not be reached directly from external networks. You must first access a login node.
```
Comment (apeck12) [inline docs/before/accounts.md:48]: Extra space in cannot

**Review by apeck12 — APPROVED** [review]
Thanks for the changes and added figures -- all looks great! Just two minor comments.

File: docs/before/accounts.md (hunk 10 above)
Comment (fredericpoitevin) [reply docs/before/accounts.md]: Thanks! Corrected.

File: docs/before/accounts.md [hunk 11]
```diff
@@ -0,0 +1,125 @@
...
+| ![Data Access](images/Data_Access.png) | 
+|:--:| 
+| *__Pathways to the Experiment.__ TBD.* |
+
+The exact pathway will depend on whether your experiment happened before 2023 or not and whether you intend to do online analyis on the FFB nodes or offline analysis. 
+We detail these different scenarii below.
+
+### Recent experiments (2023 - ...)
+
+It is anticipated that experiments carried in 2023 or after will be processed directly on S3DF.
```
[the remark this answers is not in the document]
Comment (fredericpoitevin) [reply docs/before/accounts.md]: Done

**Review by gadorlhiac — APPROVED** [review]
Looks great! Thanks Fred!
